sdb/commands/zfs/histograms.py

```diff
@@ -0,0 +1,149 @@
+    Print ZFS Histogram and print its median segment size.
+
+    NOTE
+        The median is just an approximation as we can't tell the
+        exact size of each bucket within a histogram bucket.
+
+    EXAMPLES
+        Dump the histogram of the normal metaslab class of the rpool:
+
+            sdb> spa rpool | member spa_normal_class.mc_histogram | zhist
+            idx: counts
+            ---- ------
+              9:   4264 ************************
+             10:   3168 ******************
+             11:   3545 ********************
+             12:   3344 *******************
+             13:   3724 *********************
+             14:   4184 ***********************
+             15:   7221 ****************************************
+             16:   3942 **********************
+             17:   2408 *************
+             18:   1558 *********
+             19:    979 *****
+             20:    538 ***
+             21:    390 **
+             22:    228 *
+             23:     97 *
+             24:     46
+             25:     22
+             26:      8
+             27:      9
+             28:     97 *
+            Histogram Median: 343.4MB
+    """
+
+    names = ["zfs_histogram", "zhist"]
+
+    @classmethod
+    def _init_parser(cls, name: str) -> argparse.ArgumentParser:
+        parser = super()._init_parser(name)
+        parser.add_argument("offset", nargs="?", default=0, type=int)
+        return parser
+
+    @staticmethod
+    def histogram_median(hist: drgn.Object, offset: int = 0) -> int:
+        """
+        Returns the approximated median of a ZFS histogram.
+        """
+        canonical_type = sdb.type_canonicalize(hist.type_)
+        assert canonical_type.kind == drgn.TypeKind.ARRAY
+        assert sdb.type_canonicalize(
+            canonical_type.type).kind == drgn.TypeKind.INT
+        total_space, max_bucket, min_bucket = 0, 0, (len(hist) - 1)
+
+        for (bucket, value) in enumerate(hist):
+            count = int(value.value_())
+            total_space += count << (bucket + offset)
+            if bucket < min_bucket and count > 0:
+                min_bucket = bucket
+            if bucket > max_bucket and count > 0:
+                max_bucket = bucket
+
+        space_left, median = total_space / 2, 0
+        for bucket in range(min_bucket, max_bucket + 1):
+            space_in_bucket = int(hist[bucket].value_()) << (bucket + offset)
+            if space_left <= space_in_bucket:
+                median = 1 << (bucket + offset)
+                #
+                # Size of segments may vary within one bucket thus we
+                # attempt to approximate the median by looking at the
+                # number of segments in the bucket and assuming that
+                # they are evenly distributed along the bucket's range.
+                #
+                bucket_fill_pct = (space_left * 100) / space_in_bucket
+                median += round(median * bucket_fill_pct / 100)
```

Comment:
In python3, integer division yields a floating-point result.  So I don't think we gain anything by doing this as a "percent" (`*100`).  You could do `round(median * space_left / space_in_bucket)` or
```
bucket_fill = space_left / space_in_bucket
median += round(median * bucket_fill)
```